libcloud.groovy

```diff
@@ -46,10 +48,92 @@ def replicate_to_clouds(pipecfg, basearch, buildID, stream) {
             """)
         }
     }
+    // A closure to build the aws-winli AMI. This will be called before replicating to 
+    // all regions, if the option is set for the current stream. `cosa aws-replicate` 
+    // will handle both traditional AMIs and aws-winli AMIs if present in the metadata. 
+    // aws-winli is only supported on x86_64.
+    def awsWinLIBuildClosure = { config, credentialId ->
+        def creds = [file(variable: "AWS_CONFIG_FILE", credentialsId: credentialId)]
+        withCredentials(creds) {
+            utils.syncCredentialsIfInRemoteSession(["AWS_CONFIG_FILE"])
+            def c = config
+            // In 4.19+ we switched the command to upload to clouds to
+            // `cosa imageupload-<cloud>`
+            // https://github.com/coreos/coreos-assembler/pull/4074
+            def image_upload_cmd = "imageupload"
+
+            // fetch the aws vmdk image from s3 because ore upload still requires it in the
+            // builds directory. We're already working in an initialized cosa working
+            // directory, so we only need to perform the buildfetch and decompress here.
+            def s3_stream_dir = pipeutils.get_s3_streams_dir(pipecfg, stream)
+            shwrap("""
+                cosa buildfetch --build=${buildID} \
+                --arch=x86_64 --artifact=aws --url=s3://${s3_stream_dir}/builds \
+                --aws-config-file \${AWS_CONFIG_FILE} \
+                && cosa decompress --build=${buildID} --artifact=aws
+            """)
```

Comment:
Since this is an artificial requirement in this case how about we just `touch /path/to/foo.vmdk.gz` here and then delete the empty file when we're done with the upload

Reply:
SGTM, I updated the PR to grab the file name from the metadata and then `touch` it just before creating the AMI.